api/v1alpha1/gitrepository_types.go

```diff
@@ -141,15 +155,25 @@ func GitRepositoryNotReady(repository GitRepository, reason, message string) Git
 	return repository
 }
 
+// ReadyMessage returns the message of the SourceCondition
+// of type Ready with status true if present, or an empty string.
 func GitRepositoryReadyMessage(repository GitRepository) string {
 	for _, condition := range repository.Status.Conditions {
-		if condition.Type == ReadyCondition {
+		if condition.Type == ReadyCondition && condition.Status == corev1.ConditionTrue {
 			return condition.Message
 		}
 	}
 	return ""
 }
 
+// GetTimeout returns the configured timeout or the default.
+func (in *GitRepository) GetTimeout() metav1.Duration {
```

Comment:
I think we should return `time.Duration` same as https://github.com/fluxcd/kustomize-controller/blob/master/api/v1alpha1/kustomization_types.go#L122

Reply:
Adressed.